sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs

```diff
@@ -152,6 +154,116 @@ public virtual async Task<CertificateOperation> StartCreateCertificateAsync(stri
             }
         }
 
+        /// <summary>
+        /// Creates an <see cref="X509Certificate2"/> from the specified certificate.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Because <see cref="KeyVaultCertificate.Cer"/> contains only the public key, this method attempts to download the managed secret
+        /// associated with the specified secret which contains the full certificate. If you do not have permissions to get the secret,
+        /// <see cref="RequestFailedException"/> will be thrown with an appropriate error response.
+        /// </para>
+        /// <para>
+        /// If you want a certificate containing only the public key, you can create one like:
+        /// <code snippet="Snippet:CertificateClientLiveTests_DownloadLatestCertificate">
+        /// KeyVaultCertificate certificate = await Client.GetCertificateAsync(name);
+        /// using X509Certificate2 pub = new X509Certificate2(certificate.Cer);
+        /// </code>
+        /// </para>
+        /// </remarks>
+        /// <param name="certificateName">The name of the certificate to download.</param>
+        /// <param name="version">Optional version of a certificate to download.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>An <see cref="X509Certificate2"/> from the specified certificate.</returns>
+        /// <exception cref="ArgumentException"><paramref name="certificateName"/> is empty.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="certificateName"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">The managed secret did not contain a certificate.</exception>
+        /// <exception cref="RequestFailedException">The request failed. See <see cref="RequestFailedException.ErrorCode"/> and the exception message for details.</exception>
+        public virtual Response<X509Certificate2> DownloadCertificate(string certificateName, string version = null, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(certificateName, nameof(certificateName));
+
+            using DiagnosticScope scope = _pipeline.CreateScope($"{nameof(CertificateClient)}.{nameof(DownloadCertificate)}");
+            scope.AddAttribute("certificate", certificateName);
+            scope.Start();
+
+            try
+            {
+                KeyVaultCertificateWithPolicy certificate = _pipeline.SendRequest(RequestMethod.Get, () => new KeyVaultCertificateWithPolicy(), cancellationToken, CertificatesPath, certificateName, "/", version);
```

Comment:
Do you really need this call just to get the `SecretId`? You should be able to compose the `SecretId` out of the vault url, certificate name and version.

Reply:
Practically you're right, but nothing in swagger stipulates it matches because of that. It's an observation, not a spec'd behavior. I'd rather leave it as is for now, and I'll talk with @jlichwa and @RandalliLama about clarifying in the swagger (spec). If they do, I'd be happy to come back and change this.

Reply:
Also, and I don't know how important this is, it also helps ensure they have access to the certificate (i.e. "get" permission for both cert and secret). That said, if they only had secret "get" they could do this themselves, but they could've anyway. This is just a value-add.

Reply:
It was just a suggested optimization, but you're right it would be problematic if relationship of CertificateId to SecretId were to change, since it's not called out in the spec. To your point about the permissions issue I actually think of that as more of an argument to only make the call to get the secret since that's all that really required to create the `X509Certificate2` so that's the only permission the caller should require. All that said I don't feel strongly either way, and I think it's fine to leave it as is unless the service team disagrees.